### PR TITLE
Fix/search ordering

### LINKED ISSUE #290: bug: order is not applied to search results
When using both searchQuery and order, the results are not ordered according to order.

### PR DESCRIPTION
**Pull request by dopry**
closes #290

### CODE REVIEW

**Comment by dopry** [comment]
> The changes LGTM.
> 
> My one nitpick and preference would be to have a separate testcase with the factories in `setUpTestData`.

Can do. Give me a few hours and I'll get it updated,

**Comment by dopry** [comment]
@zerolab updated!

File: grapple/utils.py [hunk 1]
```diff
@@ -127,6 +114,17 @@ def resolve_queryset(
...
+    if id is None and search_query:
+        # Check if the queryset is searchable using Wagtail search.
+        if not class_is_indexed(qs.model):
+            raise TypeError("This data type is not searchable by Wagtail.")
+
+        if grapple_settings.ADD_SEARCH_HIT:
+            query = Query.get(search_query)
+            query.add_hit()
+
+        qs = get_search_backend().search(search_query, qs)
```
Comment (zerolab) [inline grapple/utils.py:126]: So... I about to hit merge, then had my doubts (typical!)
The current version will order by relevance by default as that is the core behaviour. The PR change the logic to only use a custom order_by if that is asked for, which makes perfect sense. In fact the postgres backend does as similar thing, and more - https://github.com/wagtail/wagtail/blob/959a802c76ea760a5893e27783815e580a89d2c4/wagtail/search/backends/database/postgres/postgres.py#L586-L589 🚀

Comment (dopry) [reply grapple/utils.py:126]: What was your specific concern? If an explicit order by is provided it should be applied over relevance. Did we lose support for ordering by relevance or is it that the default behavior is changing?

Comment (zerolab) [reply grapple/utils.py:126]: all merged and in 0.19 - https://github.com/torchbox/wagtail-grapple/releases/tag/v0.19.0

Comment (dopry) [reply grapple/utils.py:126]: :tada:

**Pull request merged**